### PR TITLE
EWB-3755: Protection extensions

### PR DESCRIPTION
**Pull request by Lax125**
# Description

Adds new messages for more advanced protection modelling. See https://app.clickup.com/t/6929263/EWB-3752 for the spec.

# Associated tasks

Tasks blocking this one:
 -  [x] [Finalise model change spec](https://app.clickup.com/t/6929263/EWB-3752)

Tasks this is blocking:
- https://github.com/zepben/evolve-sdk-jvm/pull/158

# Checklist

If any of these are not applicable, strikethrough the line `~like this~`. **Do not delete it!**. Let the reviewer decide if you should have done it.

### Code
- [x] I have performed a self review of my own code (including checking issues raised when creating the PR).
- [ ] ~I have added/updated unit tests for these changes, and if not I have explained why they are not necessary.~ (Tests required downstream)
- [ ] ~I have commented my code in any hard-to-understand or hacky areas.~
- [x] I have handled all new warnings generated by the compiler or IDE.
- [x] I have rebased onto the target branch (usually main).
      
### Documentation
- [x] I have updated the changelog.
- [x] I have updated any documentation required for these changes. (protobuf comments)

# Breaking Changes
- [x] I have considered if this is a breaking change and will communicate it with other team members if so.

* Removed `ProtectionEquipment`.
  * Change of inheritance: `CurrentRelay` &rarr; `ProtectionEquipment`.
    becomes `CurrentRelay` &rarr; `ProtectionRelayFunction`.
  * Removed symmetric relation `ProtectionEquipment` &harr; `ProtectedSwitch`.
* Renamed `CurrentRelayInfo` to `RelayInfo`.
* Reworked values for enumerable type `ProtectionKind`.

### CODE REVIEW

**Comment by ryanjcoleman** [comment]
Task linked: [EWB-3752 Data model change spec](https://app.clickup.com/t/6929263/EWB-3752)

**Comment by ryanjcoleman** [comment]
Task linked: [EWB-3755 Update evolve-grpc](https://app.clickup.com/t/6929263/EWB-3755)

File: proto/zepben/protobuf/cim/iec61970/base/wires/SeriesCompensator.proto [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+   */
+  double x = 4;
+
+  /**
+   * Zero sequence reactance.
+   */
+  double x0 = 5;
+
+  /**
+   * Describe if a metal oxide varistor (mov) for over voltage protection is configured in parallel with the series
```
Comment (charlta) [inline proto/zepben/protobuf/cim/iec61970/base/wires/SeriesCompensator.proto:49]: todo: `varistorPresent` is meant to be a helper property and hence should not be included in the protobuf message.

**Review by charlta — CHANGES_REQUESTED** [review]

File: proto/zepben/protobuf/cim/iec61970/base/wires/SeriesCompensator.proto (hunk 1 above)
Comment (Lax125) [reply proto/zepben/protobuf/cim/iec61970/base/wires/SeriesCompensator.proto]: Removed `varistorPresent` and updated protolock.

**Review by charlta — APPROVED** [review]

**Review by kgreav — APPROVED** [review]

**Pull request merged**